Sources/Datadog/Tracing/TracingAutoInstrumentation.swift

```diff
@@ -0,0 +1,129 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal class TracingAutoInstrumentation {
+    static var instance: TracingAutoInstrumentation?
+
+    let swizzler: URLSessionSwizzler
+    let interceptor: RequestInterceptor
+
+    init?(tracedHosts: Set<String>) {
+        if tracedHosts.isEmpty {
+            return nil
+        }
+        do {
+            /// pattern = "^(.*\\.)*tracedHost1|^(.*\\.)*tracedHost2|..."
+            let pattern = tracedHosts
+                .map {
+                    let escaped = NSRegularExpression.escapedPattern(for: $0)
+                    return "^(.*\\.)*\(escaped)$"
+                }
+                .joined(separator: "|")
+            let regex = try NSRegularExpression(pattern: pattern, options: [.caseInsensitive, .anchorsMatchLines])
```

Comment:
Suggestion: the `TracingAutoInstrumentation` becomes more and more complex. If we want to reduce the complexity, hosts processing could be decoupled to a basic type:
```swift
let allowedHosts = AllowedHosts(hosts: ["v2.api.com"])

internal struct AllowedHosts {
    private let regex: String

    init(hosts: [String]) {}

    func matches(url: URL) -> Bool {}
}
```
This way, the `AllowedHosts` can be tested as separate unit. Currently all host-matching tests depend on the complex condition in `TracingRequestInterceptor`:
```swift
guard let tracer = Global.sharedTracer as? DDTracer,
      let someURL = urlRequest.url,
      tracedHostsRegex.allows(someURL),
      HTTPHeadersWriter.canInject(to: urlRequest) else {
         return nil
}
```
and may give false-positives due to wrong configuration.

Reply:
could be a good idea, but i don't think we need it today